tests/test_authentication.py

```diff
@@ -140,13 +139,17 @@ def test_unknown_creds(self):
                               '^access denied: CredentialsLookupError: ')
 
     def test_alternative_credential_lookup(self):
+        # Temporarily use alternate auth
+        old_auth = self.auth
+        self.auth = AlternateAuthentication()
+
         sender = self._sender(credentials=ALTERNATIVE_CREDS)
         req = self._request(sender)
-        lookup_path = "%s.alternative_lookup" % __name__
-        with self.settings(HAWK_CREDENTIALS_LOOKUP=lookup_path):
```

Comment:
In reply to https://github.com/kumar303/hawkrest/pull/34/files/a41a23a1d768ce3d5a32bf571a15d72154c7792c#r109527342

> the only advantage it brought was being able to change the setting at runtime during testing.

Yeah, to me this is a big advantage. I would like to see option 1: move the runtime check inside the default `_user_lookup()` method, with the `import_string()` overhead for every `authenticate()`. This overhead is minimal, if anything at all. The second time you import anything in Python, it is retrieved from a dict loaded into memory called `sys.modules`. This is a hash table lookup and it's insanely fast.